src/vivarium_gates_nutrition_optimization/components/hemoglobin.py

```diff
@@ -189,21 +206,53 @@ def sample_from_hemoglobin_distribution(
         ret_val.loc[gumbel] = self._mirrored_gumbel_ppf_2017(propensity, mean, sd)[gumbel]
         return ret_val
 
-    # def adjust_maternal_hemorrhage_probability(self, index, probability):
-    #     paf = self.hemorrhage_paf(index)["value"]
-    #     rr = self.hemorrhage_rr(index)["value"]
-    #     p_maternal_hemorrhage = probability["moderate_maternal_hemorrhage"] + probability["severe_maternal_hemorrhage"]
-    #     severe_ratio = probability["severe_maternal_hemorrhage"] / p_maternal_hemorrhage
-    #     p_maternal_hemorrhage_nonanemic = p_maternal_hemorrhage * (1 - paf)
-    #     p_maternal_hemorrhage_anemic = p_maternal_hemorrhage_nonanemic * rr
-    #     hemoglobin = self.hemoglobin(index)
-    #     anemic = hemoglobin <= SEVERE_ANEMIA_AMONG_PREGNANT_WOMEN_THRESHOLD
-    #     probability["severe_maternal_hemorrhage"] = severe_ratio * p_maternal_hemorrhage_nonanemic
-    #     probability["moderate_maternal_hemorrhage"] = (1 - severe_ratio) * p_maternal_hemorrhage_nonanemic
-    #     probability.loc[anemic, "severe_maternal_hemorrhage"] = severe_ratio * p_maternal_hemorrhage_anemic
-    #     probability.loc[anemic, "moderate_maternal_hemorrhage"] = (1 - severe_ratio) * p_maternal_hemorrhage_anemic
-    #     probability["not_maternal_hemorrhage"] = (1 - probability["moderate_maternal_hemorrhage"] - probability["severe_maternal_hemorrhage"])
-    #     return probability
+    def adjust_maternal_disorder_proportion(
+        self, index: pd.Index, maternal_disorder_probability: pd.DataFrame
+    ) -> pd.Series:
+        hemoglobin_level = self.hemoglobin(index)
+        rr = self.maternal_disorder_rr(index)
+        ## annoyingly formatted
+        paf = self.maternal_disorder_paf(index)["value"]
+        tmrel = TMREL_HEMOGLOBIN_ON_MATERNAL_DISORDERS
+        per_simulant_exposure = (
+            (tmrel - hemoglobin_level + abs(tmrel - hemoglobin_level)) / 2 / RR_SCALAR
+        )
+        per_simulant_rr = rr**per_simulant_exposure
+        maternal_disorder_probability *= (1 - paf) * per_simulant_rr
+        return maternal_disorder_probability.map(lambda value: 1 if value > 1 else value)
+
+    def adjust_maternal_hemorrhage_proportion(self, index, maternal_hemorrhage_probability):
+        paf = self.hemorrhage_paf(index)["value"]
+        rr = self.hemorrhage_rr(index)
+        hemoglobin = self.hemoglobin(index)
+        maternal_hemorrhage_probability *= 1 - paf
+        # Dichotomous risk based on severe anemia
+        maternal_hemorrhage_probability.loc[
+            hemoglobin <= SEVERE_ANEMIA_AMONG_PREGNANT_WOMEN_THRESHOLD
+        ] *= rr
+        return maternal_hemorrhage_probability
+
+    def adjust_hemoglobin_exposure(
```

Comment:
this is basically entirely new to sample hemorrhage severity and update hemoglobin scale factor accordingly